t/24-myjsonrpc.t

```diff
@@ -56,11 +59,28 @@ sub magic_close () {
     my $quit = myjsonrpc::read_json($isotovideo);
     is($quit, undef, "received magic close");
 }
+
 subtest magic_close => sub {
     my @warnings = warnings { magic_close() };
     like($warnings[0], qr{received magic close});
 };
 
+subtest 'send_json dies when buffer is empty and pipe is broken' => sub {
+    local *CORE::GLOBAL::syswrite = sub {
+        $! = 'broken pipe';
+        return 0;
+    };
+    dies_ok { send_json($child, undef) };
+};
```

Comment:
You can't easily override builtin functions in a different package like that.
But I think we can test this without overriding, by passing a filehandle that is not for writing:
```suggestion
subtest 'syswrite error handling' => sub {
    my @warnings = warnings {
        throws_ok {
            myjsonrpc::send_json(\*STDIN, {cmd => 'something'})
        } qr/syswrite failed: Bad file descriptor/, 'send_json dies when syswrite fails';
    };
    like $warnings[0], qr/Filehandle STDIN opened only for input/, 'got expected warning';
};

```

Reply:
Miss that and yes, I've done several tests around mocking the syswrite buildin function, but with no results.

To solve it I've extracted from the send_json function the syswrite into another function in order to mock that and it works, I've updated the code right now!

Reply:
As you seen in my latest commit:

```perl
subtest 'send_json dies when buffer is empty and pipe is broken' => sub {
    my $myjsonrpc_mock = Test::MockModule->new('myjsonrpc');
    $myjsonrpc_mock->redefine(get_written_bytes => sub ($to_fd, $json) {
            return 0;
    });
    dies_ok { myjsonrpc::send_json($child, $send1) } 'myjsonrpc: remote end terminated connection, stopping';
};
```

with get_written_bytes that is a function in myjsonrpc.pm that simply bind the syswrite buildin function:

```perl
sub get_written_bytes($to_fd, $json) {
    return syswrite($to_fd, "$json");
}
```

It is not elegant, what do you think?